h/static/scripts/annotator/anchoring/test/pdf-test.js

```diff
@@ -0,0 +1,141 @@
+'use strict';
+
+var TextQuoteAnchor = require('dom-anchor-text-quote');
+
+var FakePDFViewerApplication = require('./fake-pdf-viewer-application');
+var pdfAnchoring = require('../pdf');
+
+/**
+ * Return a DOM Range which refers to the specified `text` in `container`.
+ *
+ * @param {Element} container
+ * @param {string} text
+ * @return {Range}
+ */
+function findText(container, text) {
+  return new TextQuoteAnchor(container, text).toRange();
+}
+
+var fixtures = {
+  pdfContent: [
+    'Pride And Prejudice And Zombies\n' +
+    'By Jane Austin and Seth Grahame-Smith ',
+
+    'IT IS A TRUTH universally acknowledged that a zombie in possession of\n' +
+    'brains must be in want of more brains. Never was this truth more plain\n' +
+    'than during the recent attacks at Netherfield Park, in which a household\n' +
+    'of eighteen was slaughtered and consumed by a horde of the living dead.',
+
+    '"My dear Mr. Bennet," said his lady to him one day, "have you heard that\n' +
+    'Netherfield Park is occupied again?" ',
+  ],
+};
+
+describe('PDF anchoring', function () {
+  var container;
+  var viewer;
+
+  beforeEach(function () {
+    pdfAnchoring.purgeCache();
+
+    container = document.createElement('div');
+    document.body.appendChild(container);
+
+    window.PDFViewerApplication = viewer = new FakePDFViewerApplication({
+      container: container,
+      content: fixtures.pdfContent,
+    });
+    viewer.setCurrentPage(0);
+  });
+
+  afterEach(function () {
+    window.PDFViewerApplication.dispose();
+    window.PDFViewerApplication = null;
+    container.remove();
+  });
+
+  describe('#describe', function () {
+    it('returns position and quote selectors', function () {
+      viewer.setCurrentPage(2);
+      var range = findText(container, 'Netherfield Park');
+      return pdfAnchoring.describe(container, range).then(function (selectors) {
+        var types = selectors.map(function (s) { return s.type; });
+        assert.deepEqual(types, ['TextPositionSelector', 'TextQuoteSelector']);
+      });
+    });
+
+    it('returns a position selector with correct start/end offsets', function () {
+      viewer.setCurrentPage(2);
+      var quote = 'Netherfield Park';
+      var range = findText(container, quote);
+      var contentStr = fixtures.pdfContent.join('');
+      var expectedPos = contentStr.lastIndexOf(quote);
+
+      return pdfAnchoring.describe(container, range).then(function (selectors) {
+        var position = selectors[0];
+        var leeway = 20;
+
+        assert.closeTo(position.start, expectedPos, leeway);
```

Comment:
Originally I added this because the exact position may differ depending on how exactly the text is represented in the text layer and we're only interested in the position being roughly what is expected. However, we have enough control over the representation in the test that we don't actually need this. I'll remove it.